### DISCUSSION TITLE
Column filtering is one step behind

### DISCUSSION THREAD

**Discussion by kuvovadis** [rappasoft/laravel-livewire-tables#1562] [Q&A]
Hello,
I would ask for help with column filtering. I am trying to setup column filtering but the results are filtered by value that is previous value set in the filter. Can anyone please help?

Sample code:
`public function builder(): Builder
    {
        return User::select('*')
            ->when($this->columnSearch['name'] ?? null, fn ($query, $name) => $query->where('clients.name', 'like', '%' . $name . '%'))
            ->when($this->columnSearch['email'] ?? null, fn ($query, $email) => $query->where('clients.email', 'like', '%' . $email . '%'));
    }`

On the first search I get from $this->columnSearch['name'] empty value but after second I get first value searched.

Thanks in advance.

**Comment by lrljoe** [comment]
Known issue:
https://github.com/rappasoft/laravel-livewire-tables/issues/1491

Hoping to roll some fixes into v3 first, then backport any bug fixes into v2
